# HG changeset patch
# User Rik <rik@octave.org>
# Date 1541830085 28800
#      Fri Nov 09 22:08:05 2018 -0800
# Node ID 09665529b6bae54c4717fdf1eb769d47939ecf5e
# Parent  d02e4b1df697434a0f96ba2c24a295f06b7ab078
Add a BIST test for bitpack (bug #54931).

* typecast.cc (Fbitpack): Add a BIST test for bug #54931 where the data to be
packed is not all zeros.

diff --git a/libinterp/corefcn/typecast.cc b/libinterp/corefcn/typecast.cc
--- a/libinterp/corefcn/typecast.cc
+++ b/libinterp/corefcn/typecast.cc
@@ -437,16 +437,23 @@ column vector.
 %!assert (bitpack (zeros (1, 32,  "logical"), "uint32"), uint32 (0))
 %!assert (bitpack (zeros (1, 64,  "logical"), "int64"),  int64 (0))
 %!assert (bitpack (zeros (1, 64,  "logical"), "uint64"), uint64 (0))
 %!assert (bitpack (zeros (1, 32,  "logical"), "single"), single (0))
 %!assert (bitpack (zeros (1, 64,  "logical"), "double"), double (0))
 %!assert (bitpack (zeros (1, 64,  "logical"), "single complex"), single (0))
 %!assert (bitpack (zeros (1, 128, "logical"), "double complex"), double (0))
 
+%!test <54931>
+%! x = false (1, 32);
+%! x(1) = true;
+%! assert (bitpack (x, "uint32"), uint32 (1));
+%! x([1, 9]) = true;
+%! assert (bitpack (x, "uint32"), uint32 (257));
+
 %!error bitpack ()
 %!error bitpack (1)
 %!error bitpack (1, 2, 3)
 %!error bitpack (1, "invalid")
 %!error bitpack (1, "double")
 %!error bitpack (false, "invalid")
 %!error bitpack (false, "double")
 */
